dmriprep/interfaces/vectors.py

```diff
@@ -1,91 +1,101 @@
-#!/usr/bin/env python
-# -*- coding: utf-8 -*-
-from nipype.interfaces.base import SimpleInterface, BaseInterfaceInputSpec, TraitedSpec, File, traits, Directory, isdefined
-
-
-class CleanVecsInputSpec(BaseInterfaceInputSpec):
-    """Input interface wrapper for CleanVecs"""
-    basedir = Directory(exists=True, mandatory=False)
+"""Handling the gradient table."""
+from pathlib import Path
+import numpy as np
+from nipype.utils.filemanip import fname_presuffix
+from nipype.interfaces.base import (
+    SimpleInterface, BaseInterfaceInputSpec, TraitedSpec,
+    File, traits, isdefined
+)
+from ..utils.vectors import DiffusionGradientTable, B0_THRESHOLD, BVEC_NORM_EPSILON
+
+
+class _CheckGradientTableInputSpec(BaseInterfaceInputSpec):
     dwi_file = File(exists=True, mandatory=True)
-    fbval = File(exists=False, mandatory=False)
-    fbvec = File(exists=False, mandatory=False)
-    rasb_tsv_file = File(exists=False, mandatory=False)
-    rescale = traits.Bool(True, usedefault=True)
-    save_fsl_style = traits.Bool(True, usedefault=True)
```

Comment:
You were worried about too many things. Code a class that does just one thing, and you know 100% it does that one thing right.